### PR TITLE
[now-build-utils][now-cli] Warn instead of throwing on `api` and `pages/api`

### PR DESCRIPTION
**Pull request by AndyBitz**
When `api` and `pages/api` is used at the same we want to show a warning instead of an error and exiting.

### CODE REVIEW

**Review by leo — APPROVED** [review]

File: packages/now-build-utils/src/detect-builders.ts [hunk 1]
```diff
@@ -210,5 +211,6 @@ export async function detectBuilders(
   return {
     builders: builders.length ? builders : null,
     errors: errors.length ? errors : null,
+    warnings,
```
Comment (styfle) [inline packages/now-build-utils/src/detect-builders.ts:214]: Why are builders and errors nullable but warnings is never null?

Comment (AndyBitz) [reply packages/now-build-utils/src/detect-builders.ts:214]: Since they can always be there, builders exclude errors and errors exclude builders.

**Comment by styfle** [comment]
I think the solution to the build error is the following:

```
cd packages/now-cli
yarn remove @now/build-utils
```

We should let yarn workspaces create symlinks.

In fact, we should do this for all `@now/*` in package.json but the build script that bundles builders will need to be tweaked a bit. cc @TooTallNate

**Review by styfle — APPROVED** [review]

File: packages/now-cli/src/util/dev/builder-cache.ts [hunk 2]
```diff
@@ -39,10 +38,14 @@ const localBuilders: { [key: string]: BuilderWithPackage } = {
   },
 };
 
-const bundledBuilders = Object.keys(devDependencies).filter(d =>
+const bundledBuilders = Object.keys(pkg.devDependencies).filter(d =>
```
Comment (styfle) [inline packages/now-cli/src/util/dev/builder-cache.ts]: Is `bundledBuilders` still used? I think it should be removed.

File: packages/now-cli/src/util/dev/builder-cache.ts [hunk 3]
```diff
@@ -39,10 +38,14 @@ const localBuilders: { [key: string]: BuilderWithPackage } = {
...
   d.startsWith('@now/')
 );
 
+export function getBundledBuilders() {
+  return ['@now/go', '@now/next', '@now/node'];
```
Comment (styfle) [inline packages/now-cli/src/util/dev/builder-cache.ts]: What about `@now/python` and `@now/ruby`?

File: packages/now-cli/src/util/dev/builder-cache.ts (hunk 2 above)
Comment (styfle) [reply packages/now-cli/src/util/dev/builder-cache.ts]: Or perhaps it should be `const bundledBuilders = getBundledBuilders();`

File: packages/now-cli/package.json [hunk 4]
```diff
@@ -61,7 +61,6 @@
     "node": ">= 8.11"
   },
   "devDependencies": {
-    "@now/build-utils": "latest",
```
Comment (styfle) [inline packages/now-cli/package.json:64]: Can you also remove the other `@now/*` packages here?

File: packages/now-cli/src/util/dev/builder-cache.ts (hunk 3 above)
Comment (AndyBitz) [reply packages/now-cli/src/util/dev/builder-cache.ts]: I'm not sure, they weren't part of the package.json before, but I've added them

cc @TooTallNate

File: packages/now-cli/src/util/dev/builder-cache.ts (hunk 2 above)
Comment (AndyBitz) [reply packages/now-cli/src/util/dev/builder-cache.ts]: I replaced every `bundledBuilders` with `getBundledBuilders()`

**Pull request merged**